src/ride_type.cpp

```diff
@@ -681,18 +681,10 @@ void RideInstance::Save(Saver &svr)
 /** Default constructor of the rides manager. */
 RidesManager::RidesManager()
 {
-	std::fill_n(this->ride_types, lengthof(this->ride_types), nullptr);
-	std::fill_n(this->instances, lengthof(this->instances), nullptr);
-	std::fill_n(this->entrances, lengthof(this->entrances), nullptr);
-	std::fill_n(this->exits, lengthof(this->exits), nullptr);
 }
 
 RidesManager::~RidesManager()
 {
-	for (uint i = 0; i < lengthof(this->entrances); i++) delete this->entrances[i];
-	for (uint i = 0; i < lengthof(this->exits); i++) delete this->exits[i];
-	for (uint i = 0; i < lengthof(this->ride_types); i++) delete this->ride_types[i];
-	for (uint i = 0; i < lengthof(this->instances); i++) delete this->instances[i];
 }
```

Comment:
Constructor can be dropped. Destructor can be dropped too, it doesn't look like a baseclass at least.